pkg/local-gadget-manager/local-gadget-manager.go

```diff
@@ -0,0 +1,309 @@
+		gadgets = append(gadgets, name)
+	}
+	sort.Strings(gadgets)
+	return gadgets
+}
+
+func (l *LocalGadgetManager) ListOperations(gadget string) []string {
+	return []string{"start", "stop", "generate"}
+}
+
+func (l *LocalGadgetManager) ListTraces() []string {
+	traces := []string{}
+	for name := range l.tracers {
+		traces = append(traces, name)
+	}
+	sort.Strings(traces)
+	return traces
+}
+
+func (l *LocalGadgetManager) ListContainers() []string {
+	containers := []string{}
+	l.ContainerRange(func(c pb.ContainerDefinition) {
+		containers = append(containers, c.Id)
+	})
+	sort.Strings(containers)
+	return containers
+}
+
+func (l *LocalGadgetManager) AddTracer(gadget, name, containerFilter string) error {
+	factory, ok := l.traceFactories[gadget]
+	if !ok {
+		return fmt.Errorf("unknown gadget")
+	}
+	_, ok = l.tracers[name]
+	if ok {
+		fmt.Printf("Error: trace already exists")
+		return nil
+	}
+
+	traceResource := &gadgetv1alpha1.Trace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "gadget",
+		},
+		Spec: gadgetv1alpha1.TraceSpec{
+			Node:   "local",
+			Gadget: gadget,
+			Filter: &gadgetv1alpha1.ContainerFilter{
+				Namespace:     "local",
+				Podname:       containerFilter,
+				ContainerName: "",
+				Labels:        map[string]string{},
+			},
+			RunMode:    "Manual",
+			OutputMode: "Stream",
+		},
+	}
+
+	l.tracers[name] = tracer{
+		gadget:        gadget,
+		name:          name,
+		factory:       factory,
+		traceResource: traceResource,
+		gadgetStream:  stream.NewGadgetStream(),
+	}
+	return nil
+}
+
+func (l *LocalGadgetManager) Operation(name, opname string) error {
+	tracer, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("unknown trace")
+	}
+
+	if opname != "" {
+		gadgetOperation, ok := tracer.factory.Operations()[opname]
+		if !ok {
+			return fmt.Errorf("Unknown operation %q", opname)
+		}
+		tracerNamespacedName := tracer.traceResource.ObjectMeta.Namespace +
+			"/" + tracer.traceResource.ObjectMeta.Name
+		gadgetOperation.Operation(tracerNamespacedName, tracer.traceResource)
+	}
+
+	fmt.Printf("State: %s\n", tracer.traceResource.Status.State)
+	fmt.Printf("OperationError: %s\n", tracer.traceResource.Status.OperationError)
+	fmt.Printf("Output: %s\n", tracer.traceResource.Status.Output)
+
+	return nil
+}
+
+func (l *LocalGadgetManager) Show(name string) error {
+	return l.Operation(name, "")
+}
+
+func (l *LocalGadgetManager) Delete(name string) error {
+	tracer, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("unknown trace")
+	}
+
+	factory, ok := l.traceFactories[tracer.gadget]
+	if !ok {
+		return fmt.Errorf("unknown gadget")
+	}
+	factory.Delete("gadget/" + name)
+	delete(l.tracers, name)
+	return nil
+}
+
+func (l *LocalGadgetManager) PublishEvent(tracerID string, line string) error {
+	name := strings.TrimPrefix(tracerID, "trace_gadget_")
+	t, ok := l.tracers[name]
+	if !ok {
+		return fmt.Errorf("cannot find tracer: unknown tracer %q", tracerID)
```

Comment:
Can we use this same message in all the places to indicate a tracer doesn't exist?